src/app/modules/model/test-navigator-tree-node.ts

```diff
@@ -100,6 +100,22 @@ export class TestNavigatorTreeNode implements TreeNode {
     return newNode;
   }
 
+  /**
+   * removes this node from its parent
+   */
+  remove() {
+    if (this.parent) {
+      this.parent.removeChild(this);
+    }
+  }
+
+  removeChild(child: TestNavigatorTreeNode) {
+    if (this._children) {
```

Comment:
since this.workspaceElement.children will return this._children if existent I'd remove this special 'cache' behaviour.

Reply:
Not addressed, as discussed.